nistats/tests/test_glm_reporter.py

```diff
@@ -0,0 +1,298 @@
+                       'contrast_0, contrast_1',
+                       )
+    actual_output = glmr._make_headings(*test_input)
+    assert actual_output == expected_output
+
+
+def test_make_page_title_heading_with_contrasts_title_auto():
+    test_input = ({'contrast_0': [0, 0, 1],
+                   'contrast_1': [0, 1, 1],
+                   },
+                  'auto',
+                  )
+    expected_output = ('Report: contrast_0, contrast_1',
+                       'Statistical Report for contrasts',
+                       'contrast_0, contrast_1',
+                       )
+    actual_output = glmr._make_headings(*test_input)
+    assert actual_output == expected_output
+
+
+def test_make_page_title_heading_with_contrasts_title_custom():
+    test_input = ({'contrast_0': [0, 0, 1],
+                   'contrast_1': [0, 1, 1],
+                   },
+                  'Custom Title for report',
+                  )
+    expected_output = ('Custom Title for report',
+                       'Custom Title for report',
+                       '',
+                       )
+    actual_output = glmr._make_headings(*test_input)
+    assert actual_output == expected_output
+
+
+def test_make_page_title_heading_with_contrasts_none_title_custom():
+    test_input = (None,
+                  'Custom Title for report',
+                  )
+    expected_output = ('Custom Title for report',
+                       'Custom Title for report',
+                       '',
+                       )
+    actual_output = glmr._make_headings(*test_input)
+    assert actual_output == expected_output
+
+
+def _generate_img():
+    mni_affine = np.array([[-2., 0., 0., 90.],
+                           [0., 2., 0., -126.],
+                           [0., 0., 2., -72.],
+                           [0., 0., 0., 1.]])
+    
+    data_positive = np.zeros((7, 7, 3))
+    rng = np.random.RandomState(42)
+    data_rng = rng.rand(7, 7, 3)
+    data_positive[1:-1, 2:-1, 1:] = data_rng[1:-1, 2:-1, 1:]
+    
+    return nib.Nifti1Image(data_positive, mni_affine)
+
+
+def test_make_html_for_stat_maps_slice_z():
+    img = _generate_img()
+    stat_map_html_code = glmr._make_stat_map_svg(stat_img=img,
+                                                 threshold=4,
+                                                 alpha=0.5,
+                                                 cluster_threshold=0,
+                                                 height_control='fdr',
+                                                 bg_img=None,
+                                                 display_mode='ortho',
+                                                 plot_type='slice',
+                                                 )
+
+
+def test_make_html_for_stat_maps_slice_z():
+    img = _generate_img()
+    stat_map_html_code = glmr._make_stat_map_svg(stat_img=img,
+                                                 threshold=-4,
```

Comment:
Some tests test with legitimate args some have a more creative range. 4 & -4 are part of that, see if it raises errors or not. For `threshold`, it doesn't seem to matter, as long as it is the correct type.
For the specific test above, they are testing `plot_type` & `display_mode` args. Since the threshold value didn't matter I got creative.
Reflecting on this, this might be a cause for confusion to the next dev who works on this. I'll revert it to regular threshold values.